main.py

```diff
@@ -105,62 +105,62 @@ def compound_idx4(i, j, k, l):
 
 
 @cache
-def compound_idx2_reverse(ij):
+def uncompound_to_idx2(ij):
```

Comment:
why not `_reverse`?
if it needs to be shorter we can could it to `_rev` or `_inv`
I think it's clearer to have the inverse be the same name with a suffix rather than a prefix
(it's not exactly the inverse because the other one takes two args instead of a 2-tuple, but maybe it would be cleaner if we changed that so we can avoid unpacking)

If the goal is a shorter name, we could also use `idx2_fwd` and `idx2_rev` (or `idx2_inv`)

Reply:
I don't know. 

It was confusing to me that `compound_idx2_reverse` mean taking a compound and then returning an `idx2`.

Reply:
> I don't know. It was confusing to me that `compound_idx2_reverse` mean taking a compound and then returning an `idx2`.
> 
> Where `counpound_idx2` take a `idx2` and return a compound. Maybe it just me, I can reverse back.

I just think of it as: `compound_idx2` makes a compound index from two single indices, and `f_reverse` does the opposite of `f` (so `f(f_reverse(x)) == x` (with maybe some unpacking))

Reply:
ok let me reverse